pkg/gadgets/internal/socketenricher/tracer.go

```diff
@@ -74,7 +74,7 @@ func (se *SocketEnricher) start() error {
 
 	err = kallsyms.SpecUpdateAddresses(specIter, []string{"socket_file_ops"})
 	if err != nil {
-		return fmt.Errorf("updating socket_file_ops address with ksyms: %w", err)
+		log.Warnf("updating socket_file_ops address with ksyms: %v\nThis may result in network related information missing. Possible cause can be lack access of /proc/kallsyms due to CAP_SYSLOG not being set.", err)
```

Comment:
This can also be due to a kernel that does not provide the symbol `socket_file_ops` in `/proc/kallsyms` because the constant is declared as `static`.

That's a too long explanation to add in the `log.Warnf`. But you can add a comment before.